File: 2562-find-the-array-concatenation-value/2562-find-the-array-concatenation-value.java
class Solution {
    public long findTheArrayConcVal(int[] nums) {
        long res = 0L;
        int left = 0;
        int right = nums.length - 1;

        while (left < right) {
            int a = nums[left++];
            int b = nums[right--];
            res += a * pow10(b) + b;
        }

        if (left == right) {
            res += nums[left];
        }

        return res;
    }

    // Helper to calculate 10^digits(b)
    private long pow10(int num) {
        long pow = 1;
        while (num > 0) {
            pow *= 10;
            num /= 10;
        }
        return pow;
    }
}

File: 2562-find-the-array-concatenation-value/2562-find-the-array-concatenation-value.cpp
class Solution {
public:
    long long findTheArrayConcVal(vector<int>& nums) {
        int l = 0, r = nums.size() - 1;
        long long res = 0;
        while(l < r){
            int num = nums[r], pow = 1;
            res += num;
            while(num){
                num /= 10;
                pow *= 10;
            }
            res += nums[l] * pow;
            ++l; --r;
        }
        if(l == r) res += nums[l];
        return res;
    }
};